apps/calendar/configure.js

```diff
@@ -0,0 +1,112 @@
+/* global module, require */
+'use strict';
+var exec = require('mz/child_process').exec;
+var fs = require('mz/fs');
+var rjs = require('requirejs');
+
+module.exports = [
+  {
+    inputs: ['build/config.json'],
+    outputs: 'js/common/presets.js',
+    rule: function() {
+      return fs.readFile('./build/config.json', 'utf8').then(function(data) {
+        return fs.writeFile('./js/common/presets.js', 'define(' + data + ');');
+      });
+    }
+  },
+
+  {
+    inputs: ['build/calendar.build.js', 'js/**/*.js'],
+    outputs: [
+      'js/bundle.js',
+      'js/caldav_worker.js',
+      'js/lazy_loaded.js',
+      'js/ext/caldav.js',
+      'js/ext/ical.js'
+    ],
+    rule: function() {
+      var configFile = './build/calendar.build.js';
+      return fs.readFile(configFile, 'utf8').then(function(data) {
+        var config = JSON.parse(data);
+        return new Promise(function(resolve) {
+          rjs.optimize(config, resolve);
+        });
+      });
+    }
+  },
+
+  {
+    inputs: ['index.html'],
+    outputs: [
```

Comment:
Sometimes outputs should be decided by inputs automatically through a scanning stage so developers don't need to write down every outputs individually.

We already have do scan & copy shared resource to each app in build/webapp-shared.js. In my opinion, we could have a new plugin for parsing shared resources in html. For instance:

```
var sources = ['index.html'];
{
  inputs: sources,
  outputs: sharedParser(sources)
}